internal/cluster/query.go

```diff
@@ -0,0 +1,46 @@
+package cluster
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/buildkite/cli/v3/internal/graphql"
+	"github.com/buildkite/cli/v3/pkg/cmd/factory"
+)
+
+func QueryCluster(ctx context.Context, OrganizationSlug string, ClusterID string, f *factory.Factory) (Cluster, error) {
+
+	q, err := graphql.GetClusterQueues(ctx, f.GraphQLClient, OrganizationSlug, ClusterID)
+
+	if err != nil {
+		fmt.Println("Unable to read Cluster Queues: ", err.Error())
+		return Cluster{}, err
+	}
+
+	ClusterDescription := q.Organization.Cluster.Description
+
+	cluster := Cluster{
+		OrganizationSlug: OrganizationSlug,
+		ClusterID:        ClusterID,
+		Name:             q.Organization.Cluster.Name,
+		Description:      string(*ClusterDescription),
+		Queues:           []Queue{},
+	}
+
+	for _, edge := range q.Organization.Cluster.Queues.Edges {
+		agent, err := graphql.GetClusterQueueAgent(ctx, f.GraphQLClient, OrganizationSlug, []string{edge.Node.Id})
```

Comment:
This is good. Next up for your go learning is go-routines. You can fire off all these requests in parallel to speed up the resolution. We can kick that can down the road though

Reply:
Sure will have a look into go-routines.

Reply:
Not for this PR though. Just for your learning. And can start to implement it in places as you go